storage/mutable_segment.go

```diff
@@ -380,6 +386,30 @@ func (s *mutableSeg) collectFieldsForRawQueryWithLock(
 	fieldIndexMap = make([]int, numFieldsForQuery)
 	queryFields = make([]indexfield.DocsField, 0, numFieldsForQuery)
 
+	cleanup := func() {
+		for i := range queryFields {
+			if queryFields[i] != nil {
+				queryFields[i].Close()
+				queryFields[i] = nil
+			}
+		}
+	}
+
+	// The timestamp field and the raw doc source field must be at the head of queryFields.
+	fm := q.FieldConstraints[s.timestampFieldHash]
```

Comment:
Actually I think I have a cleaner fix in a followup PR #89 , perhaps we can get that merged first and you can rebase?

Reply:
Basically there's no requirement on maintaining this ordering, users should use `fieldIndexMap` to map the source index to the query field index, and use the mapped index to index into `queryFields`.